readthedocs/oauth/views.py

```diff
@@ -0,0 +1,557 @@
+import hmac
+from functools import cached_property
+
+import structlog
+from django.conf import settings
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from readthedocs.api.v2.views.integrations import GITHUB_EVENT_HEADER
+from readthedocs.api.v2.views.integrations import GITHUB_SIGNATURE_HEADER
+from readthedocs.api.v2.views.integrations import ExternalVersionData
+from readthedocs.api.v2.views.integrations import WebhookMixin
+from readthedocs.core.views.hooks import VersionInfo
+from readthedocs.core.views.hooks import build_external_version
+from readthedocs.core.views.hooks import build_versions_from_names
+from readthedocs.core.views.hooks import close_external_version
+from readthedocs.core.views.hooks import get_or_create_external_version
+from readthedocs.core.views.hooks import trigger_sync_versions
+from readthedocs.oauth.models import GitHubAppInstallation
+from readthedocs.oauth.services.githubapp import get_gh_app_client
+from readthedocs.projects.models import Project
+from readthedocs.vcs_support.backends.git import parse_version_from_ref
+
+
+log = structlog.get_logger(__name__)
+
+
+class GitHubAppWebhookView(APIView):
+    """
+    Handle GitHub App webhooks.
+
+    All event handlers try to create the installation object if it doesn't exist in our database,
+    except for events related to the installation being deleted or suspended.
+    This guarantees that our application can easily recover if we missed an event
+    in case our application or GitHub were down.
+    """
+
+    authentication_classes = []
+
+    @cached_property
+    def gha_client(self):
+        return get_gh_app_client()
```

Comment:
We should use `gha` or `gh_app` consistently. I'd prefer `gh_app` since I read GHA as `GH actions`.